crowdstrike.fdr: Optionally allow data deduplication (#13109)
elastic/integrations: packages/crowdstrike/_dev/build/docs/README.md

    Please see[Create Shared Credentials File](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/create-shared-credentials-file.html)
    for more details.

#### Troubleshooting

##### Duplicate Events

> The option `Enable Data Deduplication` allows you to avoid consuming duplicate events. By default, this option is set to `false` hence duplicate events may be ingested. When this option is enabled, a [fingerprint processor](https://www.elastic.co/guide/en/elasticsearch/reference/current/fingerprint-processor.html) is used to calculate hash from a set of crowdstrike fields that uniquely identifies the event. The hash is fed to Elasticsearch [`_id`](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-id-field.html) field that makes the document unique, thus avoiding duplicates.

Reviewer: ```suggestion
The option `Enable Data Deduplication` allows you to avoid consuming duplicate events. By default, this option is set to `false`, and so duplicate events may be ingested. When this option is enabled, a [fingerprint processor](https://www.elastic.co/guide/en/elasticsearch/reference/current/fingerprint-processor.html) is used to calculate a hash from a set of Crowdstrike fields that uniquely identifies the event. The hash is assigned to the Elasticsearch [`_id`](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-id-field.html) field that makes the document unique, thus avoiding duplicates.
```

Author: Done in [783e51d](https://github.com/elastic/integrations/pull/13109/commits/783e51d46a7615bd2fe6e267fb0fa442beaa658f)

> If duplicate events are ingested, to help find them, the integration `event.id` field is populated by concatenating few crowdstrike fields that uniquely identifies the event. These fields are `id`, `aid`, and `cid` from crowdstrike event. The fields are seperated with pipe `|`.

Reviewer: ```suggestion
If duplicate events are ingested, to help find them, the integration `event.id` field is populated by concatenating a few Crowdstrike fields that uniquely identifies the event. These fields are `id`, `aid`, and `cid` from the Crowdstrike event. The fields are separated with pipe `|`.
```

Author: Done in [783e51d](https://github.com/elastic/integrations/pull/13109/commits/783e51d46a7615bd2fe6e267fb0fa442beaa658f)

> For example, if your crowdsrike event contains `id: 123`, `aid: 456`, and `cid: 789` then the `event.id` is derived as `123|456|789`.

Reviewer: ```suggestion
For example, if your Crowdstrike event contains `id: 123`, `aid: 456`, and `cid: 789` then the `event.id` would be `123|456|789`.
```

Author: Done in [783e51d](https://github.com/elastic/integrations/pull/13109/commits/783e51d46a7615bd2fe6e267fb0fa442beaa658f)
